services/blockchain-connector/shared/sdk/client.js

```diff
@@ -56,22 +58,25 @@ const checkIsClientAlive = async () =>
 
 		if (
 			config.isUseLiskIPCClient ||
-			Date.now() - lastClientAliveTime < CLIENT_ALIVE_ASSUMPTION_TIME
+			Date.now() - lastClientAliveTime < CLIENT_ALIVE_ASSUMPTION_TIME ||
+			Date.now() - heartbeatCheckBeginTime < HEARTBEAT_ACK_MAX_WAIT_TIME
```

Comment:
`HEARTBEAT_ACK_MAX_WAIT_TIME` is for the duration when the `CLIENT_ALIVE_ASSUMPTION_TIME` since `lastClientAliveTime` has already passed and then the heartbeat checks are performed. `Date.now() - heartbeatCheckBeginTime < HEARTBEAT_ACK_MAX_WAIT_TIME` check was added so that not multiple pings are made to the WS server soon after everytime the `checkIsClientAlive` is invoked and this overloads the server.